common/lib/api-client/middleware/post.js

```diff
@@ -25,6 +18,10 @@ module.exports = function post(maxFileSize) {
         return payload
       }
 
+      const originalMiddleware = find(jsonApi._originalMiddleware, {
+        name: 'POST',
+      })
+
```

Comment:
Unsure how this change is related to got and if it is, does it need a feature flag?

Reply:
It's _technically_ not but I made a change to this file originally and tidied up these parts whilst I was there. So seeing as I was still working on that part of the codebase I left these changes.

The early return is related to caching. If a cached response is set the idea was to bypass the middleware. However, the `POST` middleware is never called as we never (and probably should never?) cache a `POST` response.

The other change just moves the find to only when it is needed. I figured less unnecessary computation would be better :)